src/routes/editor/[bookID]/+page.svelte

```diff
@@ -277,10 +409,188 @@
+                        submitIllustration(illustration.id, illustration.illustration)
+                        toastStore.close(progressToastId)
+                        toastStore.trigger(successUploadToast);
+                    } else if (response.error.error === "Duplicate") {
+                        uploadFileToBucket(file, bucket, ('copy_' + file.name), illustration)
+                    } else if (response.error.error === "Payload too large" || response.error.statusCode === "413"){
+                        //file too large
+                        toastStore.close(progressToastId)
+                        const errorUploadToast: ToastSettings = {
+                            message: 'File is too large. Please upload a smaller file.',
+                            // Provide any utility or variant background style:
+                            background: 'variant-filled-error',
+                        };
+                        toastStore.trigger(errorUploadToast);
+                    } else {
+                        //error
+                        toastStore.close(progressToastId)
+                        toastStore.trigger(errorUploadToast);
+                    }
+				})
+	}
+
+	/**
+	 * Uploads an illustration to supabase storage
+	 * @param newIllustrationFile - the file to upload or the event that contains the file
+	 * @param illustration
+	 */
+	async function uploadIllustration (newIllustrationFile: File | Event, illustration: Illustration) {
+
+		// if the file is an event, get the file from the event
+		if (newIllustrationFile instanceof Event) {
+			newIllustrationFile = (newIllustrationFile.target as HTMLInputElement)?.files?.[0] as File;
+			console.log('newIllustrationFile')
+		}
+
+		if (!newIllustrationFile) {
+			return; // No file selected
+		}
+
+		const chapterId = getCurrentChapter()?._id
+		const bookId = getCurrentChapter()?.book
+
+		//retrieve supabase storage bucket
+		let bucketName = `books/${bookId}/chapters/${chapterId}`
+
+		//check if bucket exists
+		supabase.storage
+				.getBucket('books')
+				.then((response: StorageBucketError) => {
+					if (response.error || !response.data) {
+						// bucket not found, create bucket first
+						supabase.storage
+								.createBucket(bucketName, {
+									public: false,
+									allowedMimeTypes: ['image/png', 'image/jpeg', 'image/gif', 'image/svg'],
+									fileSizeLimit: 1024
+								}).then(() => {
+									//upload file to bucket
+								uploadFileToBucket(newIllustrationFile as File, bucketName, undefined, illustration)
+						})
+					}
+				}).catch((error) => {
+					toastStore.trigger(errorUploadToast);
+		})
+
+		return true
+	}
+
+	/**
+	 * Gets the current chapter from the quill
+	 */
+	function getCurrentChapter(){
+		return quill.chapter
+	}
+
+	/**
+	 * Replaces the illustration src with the selected file's src
+	 * @param event
+	 */
+	function replaceIllustrationSrc(event: Event) {
```

Comment:
Could you please move this logic to quill.ts also.

Reply:
This function is only frontend logic. I have updated the comment to more accurately describe the function.

/**
* Replaces the illustration image element src with the selected file's src
* @param event
*/